File: recovery-tools/vultisig-share-decoder/cmd/generate_test_vectors.go
package main

import (
	"fmt"
	"main/internal/utils"
)

func main() {
	testCases := []struct {
		password  string
		plaintext string
	}{
		{"mypassword", "helloworld"},
		{"test123", "test data"},
		{"SecurePassword123!", "This is a longer test message with more content."},
	}

	fmt.Println("// Generated test vectors for AES-GCM decryption tests")
	fmt.Println("// Format: { password, base64EncryptedData, expectedPlaintext }")
	fmt.Println()

	for i, tc := range testCases {
		encrypted, err := utils.EncryptWithPassword([]byte(tc.plaintext), tc.password)
		if err != nil {
			fmt.Printf("Error encrypting test case %d: %v\n", i+1, err)
			continue
		}

		fmt.Printf("Test Case %d:\n", i+1)
		fmt.Printf("  Password: '%s'\n", tc.password)
		fmt.Printf("  Plaintext: '%s'\n", tc.plaintext)
		fmt.Printf("  Encrypted (base64): '%s'\n", encrypted)
		fmt.Println()
	}
}

File: recovery-tools/vultisig-share-decoder/web/generate_test_vectors.js
import crypto from 'crypto';

function encryptWithAesGcm({ key, value }) {
  const hash = crypto.createHash('sha256').update(key).digest();
  
  const nonce = crypto.randomBytes(12);
  
  const cipher = crypto.createCipheriv('aes-256-gcm', hash, nonce);
  
  const ciphertext = Buffer.concat([
    cipher.update(value),
    cipher.final()
  ]);
  
  const authTag = cipher.getAuthTag();
  
  const result = Buffer.concat([nonce, ciphertext, authTag]);
  
  return result.toString('base64');
}

const testVectors = [
  { password: 'mypassword', plaintext: 'helloworld' },
  { password: 'test123', plaintext: 'test data' },
  { password: 'SecurePassword123!', plaintext: 'This is a longer test message.' }
];

console.log('// Generated test vectors for AES-GCM decryption');
console.log('// Format: { password, base64Data, expectedText }\n');

for (const { password, plaintext } of testVectors) {
  const encrypted = encryptWithAesGcm({
    key: password,
    value: Buffer.from(plaintext, 'utf-8')
  });
  
  console.log(`{ password: '${password}', data: '${encrypted}', expected: '${plaintext}' },`);
}
